### ISSUE TITLE
"Index out of bounds" error where there are trailing spaces

### ISSUE THREAD

**Issue by vrurg** [gura-conf/gura-rs-parser#13]
This is kind of a continuation to #12. With the following string used as test input:

```rust
    let src = r#"
app_dir: "$HOME/.var/app/foo"
logging:
    console:
        level: "debug"
    file:
        dest:
            path: "$HOME/.var/log/foo.log"
        options:
            pid: true

    "#;
```

I get the following error:

```
index out of bounds: the len is 187 but the index is 187
```

A workaround is to use `src.trim_end()` since the problem is in the trailing spaces before the `"#`.

**Comment by Genarito** [comment]
Hi @vrurg , thank you very much for the report!

The problem was with some bugs when checking if I had reached the end of the file and the consideration of `\r\n` in Windows.
Fortunately, all those problems were solved and your case was added as a test to the library 🚀 

The new versions are now available with the problems solved:
- [Gura 0.5.4](https://crates.io/crates/gura/0.5.4)
- [Serde Gura 0.1.8](https://crates.io/crates/serde_gura/0.1.8) (has fixed version of Gura now)

Greetings and thanks for using Gura! 😄
